CMakeLists.txt

```diff
@@ -6,8 +6,13 @@ enable_testing()
 
 option(TRACCC_UNIT_TESTS "Enable unit tests" On)
 option(TRACCC_BENCHMARKS "Enable benchmark tests" On)
+option(TRACCC_BUILD_CUDA "Build CUDA" ON)
 
-set(CMAKE_CXX_STANDARD 17)
+if(TRACCC_BUILD_CUDA)
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
I think that instead of making this conditional on TRACCC_BUILD_CUDA, we should make it configurable with a default value irrespective of whether CUDA is enabled or not, and have the if(TRACCC_BUILD_CUDA) part check that it's set to a standard that is compatible with the CUDA code.